src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -53,7 +66,299 @@ class DataFrameWriter:
 
     def __init__(self, dataframe: "snowflake.snowpark.dataframe.DataFrame") -> None:
         self._dataframe = dataframe
+        self._cur_options: dict[str, str] = {}
+        self._copy_options: dict[str, str] = {}
         self._save_mode = SaveMode.ERROR_IF_EXISTS
+        self._partition_by = None
+        self._file_format_name = None
+        self._file_format_type = None
+        self._header = False
+        self._path = None
+
+    def save(
+        self,
+        path: str = None,
+        format: Optional[Literal["csv", "json", "parquet"]] = None,
+        mode: str = None,
+        partitionBy: Optional[ColumnOrSqlExpr] = None,
+        block=True,
+        **kwargs,
+    ) -> Union[List[Row], AsyncJob]:
+        """
+        Unloads data into the given path.
+
+        This method is typically the last method in a daisy chain of :class:`DataFrameWriter` methods.
+        It allows specifying the file type using the ``format`` parameter, providing a schema using the ``schema`` parameter,
+        and passing any copy or format-specific options using the ``kwargs`` parameter.
+
+        Alternatively, you can use the ``format`` method and specify the format, and then use the ``save`` method to specify the path:
+
+            ``df = session.read.format("csv").option("header", "true").save("path/to/your/file.csv")``
+
+        Args:
+            path: stage location for the file(s) to be loaded
```

Comment:
The API signature indicates that the path is an optional parameter. If that's true, then what's the default value?

Reply:
This can be set by `option` method but there is still confusion here. We don't check if the value is not NULL and this can lead to error without a good error message. We should improve the error checking here